core/src/main/java/org/apache/iceberg/deletes/EqualityDeleteWriter.java

```diff
@@ -39,17 +40,25 @@
   private final StructLike partition;
   private final ByteBuffer keyMetadata;
   private final int[] equalityFieldIds;
+  private final SortOrder sortOrder;
   private DeleteFile deleteFile = null;
 
   public EqualityDeleteWriter(FileAppender<T> appender, FileFormat format, String location,
                               PartitionSpec spec, StructLike partition, EncryptionKeyMetadata keyMetadata,
                               int... equalityFieldIds) {
+    this(appender, format, location, spec, partition, keyMetadata, null, equalityFieldIds);
+  }
+
+  public EqualityDeleteWriter(FileAppender<T> appender, FileFormat format, String location,
```

Comment:
Do we want to deprecate the old constructor? Not a big deal but I am bit worried about maintaining a new constructor each time we modify this class. At the same time, converting this into a builder is probably an overkill. This is a pretty internal API too so I'd be fine with breaking it.

Thoughts, @openinx @rdblue @yyanyy?

Reply:
Good point, I think we can remove the old constructor in this case, as I think this class is for supporting V2 table and thus shouldn't have a lot of dependencies outside of the library packages for production usage, so we should probably be able to do it now. I will update the PR to update the existing constructor unless people have further comments on this thread.

Reply:
I'd be fine keeping if there is a valid use case but I'd hope folks use a higher-level API for writing deletes.